src/semidiscretization/semidiscretization_hyperbolic_parabolic.jl

```diff
@@ -39,46 +39,6 @@ struct SemidiscretizationHyperbolicParabolic{Mesh, Equations, EquationsParabolic
-                                                   CacheParabolic}(mesh::Mesh,
-                                                                   equations::Equations,
-                                                                   equations_parabolic::EquationsParabolic,
-                                                                   initial_condition::InitialCondition,
-                                                                   boundary_conditions::BoundaryConditions,
-                                                                   boundary_conditions_parabolic::BoundaryConditionsParabolic,
-                                                                   source_terms::SourceTerms,
-                                                                   solver::Solver,
-                                                                   solver_parabolic::SolverParabolic,
-                                                                   cache::Cache,
-                                                                   cache_parabolic::CacheParabolic) where {
-                                                                                                           Mesh,
-                                                                                                           Equations,
-                                                                                                           EquationsParabolic,
-                                                                                                           InitialCondition,
-                                                                                                           BoundaryConditions,
-                                                                                                           BoundaryConditionsParabolic,
-                                                                                                           SourceTerms,
-                                                                                                           Solver,
-                                                                                                           SolverParabolic,
-                                                                                                           Cache,
-                                                                                                           CacheParabolic
-                                                                                                           }
-        @assert ndims(mesh) == ndims(equations)
-
-        # Todo: assert nvariables(equations)==nvariables(equations_parabolic)
-
-        performance_counter = PerformanceCounterList{2}(false)
-
-        new(mesh, equations, equations_parabolic, initial_condition,
-            boundary_conditions, boundary_conditions_parabolic,
-            source_terms, solver, solver_parabolic, cache, cache_parabolic,
-            performance_counter)
```

Comment:
This changes the behavior. Since we wrote this inner constructor, it was the only one accessible for constructing a `SemidiscretizationHyperbolicParabolic`. In particular, every `SemidiscretizationHyperbolicParabolic` will ensure that `ndims(mesh) == ndims(equations)` and that the `performance_counter` is initialized as `PerformanceCounterList{2}(false)`.

Without this inner constructor, a default constructor is added that allows people to create a `SemidiscretizationHyperbolicParabolic` without these properties - that are assumed in the code.

Reply:
Yes, exactly. It's my fault that I didn't add this explaining comment to the inner constructors - sorry for that. Could you please revert the removal of the inner constructor and add such a comment? The remaining clean-up is a good idea 👍

Reply:
@DanielDoehring Could you please revert the removal of the inner constructor?

Reply:
Great, thanks a lot!